src/box/iproto.cc

```diff
@@ -206,14 +211,29 @@ unsigned iproto_readahead = 16320;
 /* The maximal number of iproto messages in fly. */
 static int iproto_msg_max = IPROTO_MSG_MAX_MIN;
 
+static int
+iproto_fill_bound_address(char *buf, unsigned idx)
+{
+	return sio_addr_snprintf(buf, SERVICE_NAME_MAXLEN,
+				 (struct sockaddr *)
+				 	&tx_binaries[idx].addrstorage,
+				 tx_binaries[idx].addr_len);
+}
+
 const char *
 iproto_bound_address(char *buf)
 {
-	if (tx_binary.addr_len == 0)
+	if (tx_binaries_count == 0)
 		return NULL;
-	sio_addr_snprintf(buf, SERVICE_NAME_MAXLEN,
-			  (struct sockaddr *)&tx_binary.addrstorage,
-			  tx_binary.addr_len);
+	assert(tx_binaries_count <= IPROTO_LISTEN_SOCKET_MAX);
```

Comment:
I found no checks that count of binaries is less than IPROTO_LISTEN_SOCKET_MAX. So this assertion can fail

Reply:
Fixed